src/test/mir-opt/extern-static-ref.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(never_type)]
+#![feature(raw_ref_macros)]
+#![feature(raw_ref_op)]
```

Comment:
Maybe add `deny(unreachable_code)` (with a comment) to ensure even more that this test fails without the PR? The MIR output is auto-blessable so there is some risk that it'll just get blessed in the future if this ever regresses...